sys/net/application_layer/dhcpv6/client.c

```diff
@@ -494,27 +505,21 @@ static int _preparse_advertise(uint8_t *adv, size_t len, uint8_t **buf)
 static void _schedule_t2(void)
 {
     if (t2 < UINT32_MAX) {
-        uint64_t t2_usec = t2 * US_PER_SEC;
-
-        rebind_time = (xtimer_now_usec64() / US_PER_SEC) + t2;
-        xtimer_remove(&rebind_timer);
-        rebind_timer.callback = _post_rebind;
+        rebind_time = _now_sec() + t2;
```

Comment:
this and  l.: 842 seem to be the only lines that access `rebind_time`, they cancel each other  `+t2` and  `-t2`

at a first glance the `-t2` in  l.: 842 may be wrong (but might have been working around: failing renew since `mrd` seems to be used to evaluate if a renew is atempted '-t2' make this only be done if the renew is run in the same second as the "_schedule_t2").

Reply:
This PR aims not to change the logic of the DHCPv6 implementation, but only refactors, so even if it is wrong, that change would belong into a different PR. Hovewer, L. 842 calculates the MRD for a RENEW transmission (please refer to the RFC to figure out what this means exactly, the implementation was a few years ago, I would need to reread it myself, to explain it properly myself; all the more reason why I don't want to change logic in this PR ... ;-)). This line only reschedules the next T2 event which was supplied in a REPLY message by the IA_PD option (or IA_NA option, once #16228 is in), so I don't see how the two operations would be related. I guess in L. 842 we want some kind of offset to T2, as for RENEW the MRD shall be ["Remaining time until earliest T2"](https://datatracker.ietf.org/doc/html/rfc8415#section-18.2.4).

Concerns about things happening at the same time don't make sense, since the client runs in a single thread ;-).

Reply:
Something is wrong with the mrd calculation in the case of "renew" and "rebind" (request does not calculate an mrd).

(rebind will never happen, renew may run longer than t2 allows)

The error does not influence how the system works if the first renew is successful -> dificult to see in the wild

I only looked at these lines because i checked on the xtimer ztimer conversion.

The error(s) was in there before.

everything done in this PR is working as expected